oap-ape/ape-java/ape-client/src/main/java/com/intel/oap/ape/service/netty/client/NettyClient.java

```diff
@@ -45,11 +45,11 @@
 
     private Bootstrap bootstrap;
 
-    public NettyClient(int timeoutSeconds) {
+    public NettyClient(int timeoutSeconds, int nThreads) {
 
         final long start = System.nanoTime();
 
-        EventLoopGroup workerGroup = new NioEventLoopGroup();
+        EventLoopGroup workerGroup = new NioEventLoopGroup(nThreads);

```

Comment:
Do we intend to maintain a multi-thread nettyClient at executor level instead of using a single nettyClient for each task?

Reply:
Not at executor level. Just use `NettyParquetRequestHelper` in each task. It will create `nettyClient` instances.

Reply:
But why we need nThreads here? For each task, there will be a single channel conencted to server right?

Reply:
It will be set to number of logical cores if we don't set it explicitly.
`NettyParquetRequestHelper` will set 1 thread for each `NettyClient`.